src/pytest_html/scripts/datamanager.js

```diff
@@ -1,19 +1,25 @@
-const { getCollapsedCategory } = require('./storage.js')
+const { getCollapsedCategory, setCollapsedIds } = require('./storage.js')
 
 class DataManager {
     setManager(data) {
         const collapsedCategories = [...getCollapsedCategory(data.renderCollapsed)]
+        const collapsedIds = []
         const tests = Object.values(data.tests).flat().map((test, index) => {
             const collapsed = collapsedCategories.includes(test.result.toLowerCase())
+            const id = `test_${index}`
+            if (collapsed) {
+                collapsedIds.push(id)
```

Comment:
I'd avoid updating the collapsedIds collection in the map.
Instead I would do it after the construction of collapsed collection:
`const collapsedIds = collapsed.map(({id}) => id)`

Reply:
or even better: 
setCollapsedIds(collapsed.map(({id}) => id))

Reply:
I actually had it that way at first. But thought it was better to not do two iterations over the same collection.